File: problems/Jump Game/canJump.cs
public class Solution {
    public bool CanJump(int[] nums) {
        int lastPos = nums.Length - 1;
        for (int i = nums.Length - 1; i >= 0; i--) {
            if (i + nums[i] >= lastPos) {
                lastPos = i;
            }
        }
        return lastPos == 0;
    }
}

File: problems/Jump Game/canJump.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int lastIdx = n - 1;

        for (int idx = n - 1; 0 <= idx; --idx) {
            if (idx + nums[idx] >= lastIdx) {
                lastIdx = idx;
            }
        }

        return 0 == lastIdx;
    }
}
